pkg/search/postgres/common.go

```diff
@@ -518,6 +518,53 @@ func findTableInJoins(innerJoins []Join, matchTables func(join Join) bool) (int,
 	return -1, false
 }
 
+// combineDisjunction tries to optimize disjunction queries with `IN` operator when possible.
+// If not it fallbacks to combineQueryEntries
+func combineDisjunction(entries []*pgsearch.QueryEntry) *pgsearch.QueryEntry {
+	if len(entries) == 0 {
+		return nil
+	}
+	if len(entries) == 1 {
+		return entries[0]
+	}
+
+	exactQuerySuffix := " = $$"
+
+	seenQueries := set.StringSet{}
+	seenSelectFields := set.StringSet{}
+	values := make([]any, 0, len(entries))
+	for _, entry := range entries {
+		if entry.Having != nil ||
+			len(entry.GroupBy) != 0 ||
```

Comment:
Its a little late, but some comments in here would have been helpful.  For instance why does the existence of `having` or `group by` matter?

Reply:
https://github.com/stackrox/stackrox/pull/14127